packages/core/src/components/drawer/drawer.tsx

```diff
@@ -120,34 +127,50 @@ export class Drawer {
     }
   }
 
-  private slideOutRight(el?: HTMLElement) {
-    if (el) {
-      anime({
-        targets: el,
-        duration: Drawer.duration,
-        translateX: [0, '16rem'],
-        opacity: [1, 0],
-        easing: 'easeInSine',
-        complete: () => {
-          el.classList.add('d-none');
-        },
-      });
-    }
+  private getConstrainedWidth(width: number) {
+    return Math.min(Math.max(width, this.minWidth), this.maxWidth);
   }
 
-  private slideInRight(el?: HTMLElement) {
-    if (el) {
-      anime({
-        targets: el,
-        duration: Drawer.duration,
-        translateX: ['16rem', 0],
-        opacity: [0, 1],
-        easing: 'easeOutSine',
-        begin: () => {
-          el.classList.remove('d-none');
-        },
-      });
-    }
+  private slideOutRight(el: HTMLElement) {
+    const initialWidth = `${this.getConstrainedWidth(
+      this.width === 'auto' ? this.minWidth : this.width
+    )}rem`;
+
+    anime({
+      targets: el,
+      duration: Drawer.duration,
+      width: [initialWidth, 0],
+      opacity: [1, 0],
+      easing: 'easeInSine',
+      begin: () => {
+        el.style.opacity = '0';
```

Comment:
Why is this part of `begin`? This will vanish the content before animation is completed. It also causes the height of the drawer to jump to ~0 when not in full height mode.